app/src/main/java/de/danoeh/antennapod/adapter/EpisodeItemListAdapter.java

```diff
@@ -5,43 +5,48 @@
 import android.view.MenuInflater;
 import android.view.View;
 import android.view.ViewGroup;
+
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.recyclerview.widget.RecyclerView;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
 import de.danoeh.antennapod.R;
 import de.danoeh.antennapod.activity.MainActivity;
 import de.danoeh.antennapod.core.feed.FeedItem;
 import de.danoeh.antennapod.core.util.FeedItemUtil;
 import de.danoeh.antennapod.fragment.ItemPagerFragment;
 import de.danoeh.antennapod.menuhandler.FeedItemMenuHandler;
 import de.danoeh.antennapod.view.viewholder.EpisodeItemViewHolder;
-import org.apache.commons.lang3.ArrayUtils;
-
-import java.lang.ref.WeakReference;
-import java.util.ArrayList;
-import java.util.List;
 
 /**
  * List adapter for the list of new episodes.
  */
-public class EpisodeItemListAdapter extends RecyclerView.Adapter<EpisodeItemViewHolder>
+public class EpisodeItemListAdapter extends SelectableAdapter<EpisodeItemViewHolder>
         implements View.OnCreateContextMenuListener {
 
     private final WeakReference<MainActivity> mainActivityRef;
     private List<FeedItem> episodes = new ArrayList<>();
+    // Used to enable O(1) search. Can't use FeedItem because it does not override equals()
+    private Set<Long> selectedIds = new HashSet<>();
+    private List<FeedItem> selectedItems = new ArrayList<>();
     private FeedItem selectedItem;
+    int selectedPosition = 0; // used to init actionMode
```

Comment:
Maybe another name here? "Select" now is used for both, the item that opened the context menu and the list of checked items.

Reply:
Changed to initialItemPosition. Also added another comment for further clarification.